Explorer/Assets/DCL/Infrastructure/SceneRuntime/Apis/Modules/RestrictedActionsApi/RestrictedActionsAPIWrapper.cs

```diff
@@ -24,16 +24,31 @@ public RestrictedActionsAPIWrapper(IRestrictedActionsAPI api, CancellationTokenS
         public bool OpenExternalUrl(string url) =>
             api.TryOpenExternalUrl(url);
 
+        private CancellationTokenSource? movePlayerToCancellationToken;
+
         [UsedImplicitly]
-        public void MovePlayerTo(
+        public object MovePlayerTo(
             double newRelativePositionX, double newRelativePositionY, double newRelativePositionZ,
             double? cameraTargetX, double? cameraTargetY, double? cameraTargetZ,
-            double? avatarTargetX, double? avatarTargetY, double? avatarTargetZ)
+            double? avatarTargetX, double? avatarTargetY, double? avatarTargetZ,
+            double? duration)
         {
-            api.TryMovePlayerTo(
-                new Vector3((float)newRelativePositionX, (float)newRelativePositionY, (float)newRelativePositionZ),
-                cameraTargetX.HasValue && cameraTargetY.HasValue && cameraTargetZ.HasValue ? new Vector3((float)cameraTargetX, (float)cameraTargetY, (float)cameraTargetZ) : null,
-                avatarTargetX.HasValue && avatarTargetY.HasValue && avatarTargetZ.HasValue ? new Vector3((float)avatarTargetX, (float)avatarTargetY, (float)avatarTargetZ) : null);
+            movePlayerToCancellationToken = movePlayerToCancellationToken.SafeRestart();
+            return MovePlayerToAsync(movePlayerToCancellationToken.Token).ToDisconnectedPromise(this);
+
+            async UniTask<bool> MovePlayerToAsync(CancellationToken ct)
+            {
+                try
+                {
+                    return await api.TryMovePlayerToAsync(
+                        new Vector3((float)newRelativePositionX, (float)newRelativePositionY, (float)newRelativePositionZ),
+                        cameraTargetX.HasValue && cameraTargetY.HasValue && cameraTargetZ.HasValue ? new Vector3((float)cameraTargetX, (float)cameraTargetY, (float)cameraTargetZ) : null,
+                        avatarTargetX.HasValue && avatarTargetY.HasValue && avatarTargetZ.HasValue ? new Vector3((float)avatarTargetX, (float)avatarTargetY, (float)avatarTargetZ) : null,
+                        duration.HasValue ? (float)duration.Value : 0f,
+                        ct);
+                }
+                catch (Exception) { return false; }
```

Comment:
It looks like this catch is redundant as you guarantee an exception-free flow by returning a boolean, and the implementation itself tells that it's exception-free